V2.16-RUN-13108-SSO-user-visibility (#614)
run-ai/docs: docs/admin/admin-ui-setup/admin-ui-users.md

---
title: Adding, Updating and Deleting Users
summary: This article describes how to add, update and delete users in the Run:ai platform.
authors:
    - Jason Novich
date: 2023-Dec-28

> ---

Reviewer: @jasonnovichRunAI should this section be in a comment?

## Introduction

> The Run:ai UI allows the creation of Run:ai **local** users. Local users are users that are only in the Run:ai platform. Users are assigned levels of access to all aspects of the UI including submitting jobs on the cluster.

Reviewer: Maybe the introduction sentences should explain that Run:ai allows the admin to manage users (not just creation), and explain what are the 2 types of users - local and SSO. 
This will set the tone for the rest of the article where we mention differences between SSO and local

> The Run:ai UI allows the creation of Run:ai **local** users. Local users are users that are only in the Run:ai platform. Users are assigned levels of access to all aspects of the UI including submitting jobs on the cluster.

Reviewer: @jasonnovichRunAI ping regarding the comments above

!!! Tip
    It is possible to connect the Run:ai UI to the organization's directory and use single sign-on (SSO). This allows you to set Run:ai roles for users and groups from the organizational directory. For further information see [single sign-on configuration](../runai-setup/authentication/sso.md).

## Working with Users

> You can create local users, as well as update and delete local users.

Author: @gal-revach my goal was to make the users "equal" in terms of actions you can perform hoping that this would save repeating SSO and Local all the time. If this is not clear, I will rewrite.

> You can create local users, as well as update and delete local users.

Reviewer: @jasonnovichRunAI I think we should somehow explain the difference. Maybe if the first sentences of the article will explain the difference, then it will be clearer in all the other parts (see my comments on line 5)

### Create a User

!!! Note

    * To be able to review, add, update and delete users, you must have *System Administrator* access. To upgrade your access, contact a system administrator.

>     * When SSO is enabled, users created using the procedure below will be local users only. You cannot create new SSO users. 

Author: ```suggestion
    * When SSO is enabled, users created using the procedure below will be local users only. You can only provide SSO users with access to the system. You cannot create new SSO users.
```

>     * When SSO is enabled, users created using the procedure below will be local users only. You can only provide SSO users with access to the system. You cannot create new SSO users.

Reviewer: @jasonnovichRunAI in the last sentence maybe we should write "You shouldn't create new SSO users" instead of "You can't create new SSO users"?

> 4. Review the new user information and note the temporary password that has been assigned. To send the user an introductory email, select the checkbox.

Author: @gal-revach I am not sure it is needed. The button changes based on the SSO flag. I can add it, it would be something like this: Press the *New User* or *New Local User* and enter... WDYT?

> 4. Review the new user information and note the temporary password that has been assigned. To send the user an introductory email, select the checkbox.

Reviewer: @jasonnovichRunAI no, I agree, let's leave it as is.
However, we should remove the sentence "To send the user an introductory email, select the checkbox.", this wasn't developed

5. Press *Done* when complete.

## Users Table

The *Users* table displays a list of users who are authorized to login to the Run:ai platform.

!!! Note
    SSO users who have not logged into the Run:ai platform will not appear in the table until they do.

> The *Users* table contains informtion about users who have logged into the Run:ai Platform. The table includes columns for:

Reviewer: @jasonnovichRunAI this is not accurate. SSO users will be displayed once they log in (as you also wrote in the note in line 40, but local users are displayed once you create them.
So maybe we should change the sentence to "The users table contains information about users in the Run:ai platform"?

> * **User**&mdash;the email and user ID of the user.

Reviewer: ```suggestion
* **User**&mdash;the email address which is the user identifier.
```

* **Type**&mdash;the type of user (local or SSO).
* **Last login**&mdash;the last time the user logged into the platform.
* **Access rule(s)**&mdash;press *VIEW* to view the access rule(s) assigned to the user.
* **Created by**&mdash;indicates who created the user in the able.
* **Creation time**&mdash;the timestamp for when the user was created.

> * **Last updated**&mdash;the last time the user information was updated.

Reviewer: @jasonnovichRunAI "Created by", "Created time", and "Last updated" will be empty for SSO users, do you think we should mention it?

Author: @gal-revach We can. It may look something like:
```suggestion
* **Last updated**&mdash;the last time the user information was updated. Not applicable to SSO users.
```
WDYT?

Reviewer: @jasonnovichRunAI yes, let's add it to the relevant fields

When you press *View* in the *Access rule(s)* column, a pop-up will appear that displays the rules assigned to the user. In the popup are the following columns:

* **Role**&mdash;the name of the role assigned to the user.
* **Scope**&mdash;the scope to which the user has rights.

> * **Type**&mdash;the type of user (SSO or other)

Reviewer: @jasonnovichRunAI 
```suggestion
* **Type**&mdash;the type of subject assigned to the access rule (User or SSO group)
```

> * **Group**&mdash;the group to which the user is assigned. (Applies only to SSO users)

Reviewer: @jasonnovichRunAI 
```suggestion
* **Group**&mdash;if the access rule is inherited from an SSO group, the name of the group assigned to the access rule (Applies only to SSO users).
```

Author: ```suggestion
* **Group**&mdash;the group to which the user is assigned. If the access rule is inherited from an SSO group, the name of the group assigned to the access rule (Applies only to SSO users).
```
@gal-revach WDYT?

Reviewer: @jasonnovichRunAI we still don't need the first sentence "the group to which the user is assigned" because it is inaccurate and confusing... maybe we should write "the name of the group assigned to the access rule, in case the access rule is inherited from an SSO group (applies only to SSO users)."
WDYT?

> * **Athorized by**&mdash;the user who granted the access roles.

Reviewer: @jasonnovichRunAI 
typo - "Authorized by"

Reviewer: @jasonnovichRunAI also - "the access rule" instead of "the access roles"

> * **Creation time**&mdash;the timestamp for when the user was created.

Reviewer: @jasonnovichRunAI should be "the timestamp for when the access rules was created"

> * **Last updated**&mdash;the last time the user information was updated.

Reviewer: @jasonnovichRunAI should be "the last time the access rules was updated"

## Assigning access rules to users

Once you have created the users you can assign them *Access rules*. This provides the needed authorization to access system assets and resources.

### Roles and permissions

Roles provide a way for administrators to group and identify collections of permissions that administrators assign to [subjects](../runai-setup/access-control/rbac.md#subjects). Permissions define the actions that can be performed on managed entities. The [Roles](../runai-setup/access-control/rbac.md#roles) table shows the default roles and permissions that come with the syste. See [Role based access control](../runai-setup/access-control/rbac.md) for more information.

> To add an *Access rule* to a user:

Reviewer: @jasonnovichRunAI 
Maybe we should also mention that Access rules can be added via he access rules page? because if an SSO user still didn't login, the admin won't see it in the Users page, but they can still add access rules to these users

Reviewer: @jasonnovichRunAI ping on my last comment

Also, maybe we should write "To add or edit an Access rule" to a user:" instead of just "add"

> Roles provide a way for administrators to group and identify collections of permissions that administrators assign to [subjects](../runai-setup/access-control/rbac.md#subjects). Permissions define the actions that can be performed on managed entities. The [Roles](../runai-setup/access-control/rbac.md#roles) table shows the default roles and permissions that come with the syste. See [Role based access control](../runai-setup/access-control/rbac.md) for more information.

Reviewer: @jasonnovichRunAI type - "...with the system" instead of "...with the syste"
